### DISCUSSION TITLE
Ramps 1.6

### DISCUSSION THREAD

**Discussion by boxgame9** [fra589/grbl-Mega-5X#189] [Q&A] [answered]
Hi Gauthier
Can you confirm that mega 5x works with the ramps 1.6 board please

Pouvez-vous confirmer que mega 5x fonctionne avec la carte des rampes 1.6 s'il vous plaît

Regards Lesli

**Comment by fra589** [comment, marked as answer]
Hi Lesli,  
  
Yes, it work with RAMPS 1.6.
J'utilise une carte RAMPS 1.6 sur l'une de mes plateformes de test.
  
@++;  
Gauthier.

**Reply by JClaudeDUSS** [reply]
Bonjour @JClaudeDUSS, Note: Merci de ne pas squatter cette discussion qui, a la base, n'a rien a voir avec la base de cette discussion. Pour tout nouveau sujet, après avoir bien cherché s'il est vraiment nouveau ([par exemple avec Google](https://www.google.com/search?q=grbl-mega-5x+square+%2Bsite%3Agithub.com%2Ffra589%2Fgrbl-Mega-5X&ei=cp5_Y8jfEKLHkdUPw_K30A4&ved=0ahUKEwjI4Mz7oMf7AhWiY6QEHUP5DeoQ4dUDCA4&uact=5&oq=grbl-mega-5x+square+%2Bsite%3Agithub.com%2Ffra589%2Fgrbl-Mega-5X&gs_lcp=Cgxnd3Mtd2l6LXNlcnAQA0oECEEYAUoECEYYAFDfCVj5M2DGRGgBcAB4AIABNYgB_gSSAQIxM5gBAKABAcABAQ&sclient=gws-wiz-serp)) merci de poser ta question dans ta propre discussion à laquelle je me ferai un plaisir de répondre.
En attendant, je masque celle ci pour ne pas polluer ce sujet.

Bonjour,
je recherche une solution pour palier au défaut de géométrie du chassi d'une CN (perpendicularité des axes).
Sur le imprimantes 3d avec Marlin, il y a la fonction "Skew Factor" que l'on configure dans le Firmware instalé sur l'arduino.
Existe t'il quelque chose d'équivalent pour grbl ?
Cordialement.
<img width="657" alt="skew factor" src="https://user-images.githubusercontent.com/50746997/203825027-e6a465d9-7a2b-4dc9-8d82-50ac237399f8.png">

**Comment by boxgame9** [comment]
Thanks for the quick reply Gauthier.
Merci pour la réponse rapide Gauthier.
Stay safe and a good weekend
Restez en sécurité et passez un bon week-end

**Comment by boxgame9** [comment]
Hi Gauthier,
Bonjour Gauthier
Although I think it unlikely will the mega 5x work on the ARM based mega 2560 please.
Bien que je pense qu'il est peu probable que le méga 5x fonctionne sur le méga 2560 basé sur ARM, s'il vous plaît.

Have a great weekend.
Passe un bon weekend.

Lesli

**Comment by fra589** [comment]
Hi Lesli,  
  
I think you want to tell about 32 bits Arm based microcontroller with the same form factor than the Arduino mega 2560 which is the Arduino Due...
  
grbl-Mega-5x is not compatible with the Arduino Due.  
  
A while ago I was thinking of porting grbl-Mega-5X to the arduino Due, but I gave up on that idea for many reasons.  
There is many other 32 bits Grbl versions, and I like the idea of keeping the Mega 2560 card running. It's a simple card, really cheap and easy to get hold of. It's ideal for beginners and good enough for the hobbyist that I am.  
  
@++;  
Gauthier.

**Comment by JClaudeDUSS** [comment]
Bonjour,
existe t'il sur Grbl Mega 5X, une fonction , 
comme sur Marlin pour les imprimantes 3d,
pour corriger les défauts de perpendicularité des axes machine ?
(Exemple de la fonction sur Marlin en pièce jointe).
Cordialement.
<img width="657" alt="skew factor" src="https://user-images.githubusercontent.com/50746997/205585306-68dcccf3-88e5-4fa9-b9ac-d0d2405f5317.png">

**Reply by fra589** [reply]
Bonjour Jean-Claude,  
  
Comme je l'ai indiqué dans ton précédent commentaire que j'ai masqué, il n'est pas de bon ton de squatter un fil de discussion qui n'a rien à voir avec ton besoin. Merci d'ouvrir ton propre fil de discussion dans un tel cas.

Cette fonction n'existe pas dans Grbl(-Mega-5X).  
L’équerrage des axes est possible actuellement par l'utilisation d'axes clonés en réglant mécaniquement les position des commutateurs de fin de course. La possibilité de réaliser ce réglage de manière logicielle est en cours de développement, j'ai commencé, mais je ne peux pas donner de délai aujourd'hui.  
Hé oui, comme tout bon retraité, je suis bien overbooké ! :smiley:  
  
@++;  
Gauthier.

**Reply by JClaudeDUSS** [reply]
Bonjour, 
et merci pour la réponse.
J'avais posté ma question ici, parce que ma carte est une Rampe 1.6.
Je vous prie de m'excuser pour la maladresse.
Ayant continué mes recherches de solution

Si cela peut aider, sur Marlin la fonction est : M852

sinon, correction manuelle qui ne fonctionne que pour les déplacements linéaires :
Quelque chose comme f(y) = y + (x*.005)
Lorsque vous êtes à X= 0, la valeur Y sera inchangée.
Lorsque vous passez à X = 500, le Y augmentera de 2,5 mm (500 * 0,005). 

Cordialement.

**Comment by Leinad1776** [comment]
Dear Gauthier,
Is there anything to be set or configured upon flashing grbl Mega 5x onto Arduino Mega for it to run with Ramps 1.6+?
I am wondering how grbl running on the Mega will communicate with Ramps 1.6+ properly.
Thank you in advance.
Leinad

**Reply by fra589** [reply]
RAMPS 1.6 have the same pinout than older RAMPS cards, so they work as well as the same manners with the same parameters.

**Reply by Leinad1776** [reply]
Thank you very much for your reply!
Kindest regards
Leinad
